### PR TITLE
projects: Add jetbrains ide plugin project

### PR DESCRIPTION
**Pull request by gaocegege**
New project: coala plugin for jetbrains IDEs

### Related issues

https://github.com/coala/coala/issues/2019

#### Chat in [gitter.im/coala/coala/gsoc](https://gitter.im/coala/coala/gsoc)
```text
Ce Gao @gaocegege 17:56
Hi, could we add coala/coala#2019 to projects.coala.io? There are many people use jetbrains IDEs.  It makes sense.

Lasse Schuirmann @sils 17:57
yeah certainly, we'd also have a comentor from jetbrains, can you make a project? I'll contact him

Ce Gao @gaocegege 17:58
Of course

Lasse Schuirmann @sils 17:59
contacted jetbrains, check. We might well get a comentor from them
```
Signed-off-by: Ce Gao <ce.gao@outlook.com>

### CODE REVIEW

**Comment by gaocegege** [comment]
There would be a co-mentor from jetbrains.

**Comment by sils** [comment]
@Adrianzatreanu do you want to do this as a primary mentor?

**Comment by Adrianzatreanu** [comment]
@sils no, i dont have any idea how to write a plugin :/

**Comment by sils** [comment]
the student will find out, you just have to make sure he does things and
guide him on googling the right thing, a mentor isn't supposed to do the
thinking for the student ;)

Sincerely,

Lasse Schuirmann

lasse@schuirmann.net
http://coala.io/ - http://viperdev.io/ - http://gitmate.com/

File: data/projects/jetbrains-ide-plugin.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+* A working plugin with minimal functionality exists.
+* A basic testsuite exists.
+* Basic documentation exists.
+
+##### GSOC 2017 FINAL
+
+* A working plugin with full functionality exists.
+* A full covering testsuite exists.
+* Documentation exists.
```
Comment (sims1253) [inline data/projects/jetbrains-ide-plugin.md]: Add emphasis on the "full" part of documentation.

**Comment by sils** [comment]
hey so the jetbrains people would like to see not only coala but also mypy and pylint *directly* integrated with their ide as part of this project, does this sound fair? They'd be providing two mentors, primary and secondary, and potentially fund a student for the project as part of their own program. IMO we should adapt the description to that.

**Comment by gaocegege** [comment]
I think mypy and pylint are not our area. 🤔

File: data/projects/jetbrains-ide-plugin.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+##### GSOC 2017 CODING STARTS
+
+* A concept exists for the project.
+* An official repository set up.
+
+##### GSOC 2017 MIDTERM
+
+* A working plugin with minimal functionality exists.
+* A basic testsuite exists.
+* Basic documentation exists.
```
Comment (gaocegege) [inline data/projects/jetbrains-ide-plugin.md]: How about `corresponding test cases exists and corresponding documentation exists`

**Comment by sils** [comment]
well we have integrations already, if we make a coala plugin we could make a mypy and pylint one while we're at it. They'd be fine if the project focuses on coala IIUC

**Comment by gaocegege** [comment]
You mean this way? 



or this way



Yeah of course the pattern is same but need some ad-hoc code to fit mypy and pylint. It's not difficult.

File: data/projects/jetbrains-ide-plugin.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#### Milestones
+
+##### GSOC 2017 CODING STARTS
+
+* A concept exists for the project.
+* An official repository set up.
+
+##### GSOC 2017 MIDTERM
+
+* A working plugin with minimal functionality exists.
```
Comment (gaocegege) [inline data/projects/jetbrains-ide-plugin.md]: Should we define it? I think it's the student's work.

**Comment by traff** [comment]
Hi, I am PyCharm team lead. I can be a co-mentor for this project. The idea behind the support of mypy and pylint along with the coala is that it's better to have a general mechanism of highlighting code analysis results from external tools. (By the way, we already have something similar for PEP8) Of course the primary goal can be to support coala, but the API should be done in a way to be able to support others.

**Comment by gaocegege** [comment]
Hi, I'm a newcomer to coala and I'm interested in this project.

So jetbrains IDEs don't have a good support to static analysis and the community want students to find a general way to write static analysis plugin?

Could we reference https://github.com/Microsoft/language-server-protocol?

File: data/projects.js [hunk 4]
```diff
@@ -379,5 +379,23 @@ coala should support generating metrics for your code.",
...
+		"requirements" : [
+				"The applicant should have at least one patch accepted to any of the coala \
+				plugins.",
+				"The applicant shoule be familiar with jetbrains IDEs, e.g. PyCharm, IntelliJ IDEA."
+		],
+		"difficulty" : "high",
+		"issues" : [
+			"https://github.com/coala/coala/issues/2019"
+		],
+		"mentors" : [/*There is a co-mentor from jetbrains*/],
```
Comment (sils) [inline data/projects.js]: please add @traff here

Comment (sils) [inline data/projects.js]: @traff who would be comentor?

File: data/projects.js [hunk 5]
```diff
@@ -379,5 +379,23 @@ coala should support generating metrics for your code.",
...
+		"initiatives" : ["GSoC", "RGSoC"],
```
Comment (sils) [inline data/projects.js]: @traff if you want we can add your program here too, just name it

File: data/projects.js (hunk 4 above)
Comment (gaocegege) [inline data/projects.js]: OK

Comment (gaocegege) [inline data/projects.js]: We have a co-mentor, so should we have a primary mentor in coala?

**Comment by sims1253** [comment]
can you polish this asap so we have it ready before the application deadline?

**Comment by gaocegege** [comment]
oh, there is only one mentor, it doesn't matter?

Then I'll update and push.

**Comment by sils** [comment]
for now it's fine. One rebase to go.

**Comment by sils** [comment]
ack https://github.com/coala/projects/pull/139/commits/b2ab1420d5857b91297e8102eb69f55b87a390c3

**Comment by sils** [comment]
@rultor merge

**Pull request merged**